keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -51,34 +56,46 @@ class ImageService(
    *
    * If no images at all exist for [artifactName] this method returns `null`.
    */
-  suspend fun getLatestImage(artifactName: String, account: String): Image? {
-    val candidateImages = cloudDriverService
-      .namedImages(DEFAULT_SERVICE_ACCOUNT, artifactName, account)
+  suspend fun getLatestImage(artifact: DeliveryArtifact, account: String): Image? {
+    val candidateImagesToVersions = cloudDriverService
+      .namedImages(DEFAULT_SERVICE_ACCOUNT, artifact.name, account)
       .filter { it.hasAppVersion && it.hasBaseImageVersion }
-      .sortedWith(NamedImageComparator)
-    val latest = candidateImages
-      .firstOrNull {
-        // TODO: Frigga and Rocket version parsing are not aligned. We should consolidate.
-        it.appVersion.parseAppVersion().packageName == artifactName
+      .associateBy { image ->
+        image.toArtifactVersion()
```

Comment:
@robfletcher Calling your attention to this change. This is now resorting to looking up the artifact version details from our database, and ignoring versions which we haven't stored. The only place this function is called is indirectly in `ImageHandler` to determine the latest baked AMI for the artifact. I'm afraid this might break that behavior because we'd ignore images for which we don't have an artifact version record in the database. Thoughts?

Reply:
On second thought, I think it's safe to keep sorting the images by version, since the point here is just to find the latest AMI for a given artifact version. I'll undo this.